### PR TITLE
fix: Style dictionary helpers

### PR DESCRIPTION
**Pull request by CCAyl**
Our custom.js file (custom helpers for style dictionary) was getting out of hand. this breaks up it up into different folders
```
helpers
    style-dictionary
        filters.js
        formats.js
        transforms.js
```

and adds some helpful comments as well.

Do we like the structure? Should we break it down even more?

no changes should be made to the tokens

### CODE REVIEW

**Review by alexandra-lim — COMMENTED** [review]
Nice, thanks for doing this! Just a few suggestions to further scale this

File: src/helpers/style-dictionary/transforms.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+StyleDictionary.registerTransform({
+  name: "custom/value/font-weight-to-string",
+  type: "value",
+  matcher: (token) => token.type === "fontWeights" || token.type === "fontWeight",
+  transformer: function (token) {
+    return token.value.toString();
+  },
+});
+
+// TRANSFORM GROUPS, used in build.js
```
[the remark this answers is not in the document]
Comment (iryanclarke) [reply src/helpers/style-dictionary/transforms.js:60]: +1 to this!

File: src/helpers/style-dictionary/transforms.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+StyleDictionary.registerTransformGroup({
+  name: "custom/native",
+  transforms: [
+    "attribute/cti",
+    "custom/name/remove-desktop-prefix",
+    "custom/value/rm-px",
+    "custom/value/font-weight-to-string",
+    "custom/name/remove-color-prefix",
+  ],
+});
```
Comment (iryanclarke) [inline src/helpers/style-dictionary/transforms.js:89]: I think we should also specify the known consumers of specific transform groups, just for future visibility, and any caveats (particularly with Native)

Comment (CCAyl) [reply src/helpers/style-dictionary/transforms.js]: added!

**Comment by iryanclarke** [comment]
LGTM!

**Review by alexandra-lim — APPROVED** [review]
Nice, thanks for doing this

**Pull request merged**